soc/arm/nordic_nrf/nrf51/soc_power.h

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright (c) 2017 Intel Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef _SOC_POWER_H_
+#define _SOC_POWER_H_
+
+#include <stdbool.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+enum power_states {
+#ifdef CONFIG_SYS_POWER_LOW_POWER_STATE
+# ifdef CONFIG_SYS_POWER_STATE_CPU_LPS_SUPPORTED
+	SYS_POWER_STATE_CPU_LPS,	/* Not used */
```

Comment:
/* not used */ is kind of misleading. `not supported by SoC`?
Why not keep enum clean without ifdefs?

Reply:
If I do so, user will be not able to add custom power modes to the board and/or application without editing this file.

Reply:
it seems that it goes back to moving this enum to the subsystem api file (if `sys_is_valid_power_state` is moved it must go too) then it has to contain all zephyr supported power modes.